lib/spack/spack/solver/concretize.lp

```diff
@@ -196,6 +218,28 @@ attr(Name, A1, A2, A3, A4) :- impose(ID), imposed_constraint(ID, Name, A1, A2, A
    not imposed_constraint(Hash, "node_flag", Package, FlagType, Flag),
    internal_error("imposed hash without imposing all flag values").
 
+% associated conditions by cause -> effect
+condition_cause(Effect, Cause) :-
+  condition_holds(Effect), condition_holds(Cause),
+  attr(Name, A1),
+  condition_requirement(Effect, Name, A1),
+  imposed_constraint(Cause, Name, A1).
+condition_cause(Effect, Cause) :-
+  condition_holds(Effect), condition_holds(Cause),
+  attr(Name, A1, A2),
+  condition_requirement(Effect, Name, A1, A2),
+  imposed_constraint(Cause, Name, A1, A2).
+condition_cause(Effect, Cause) :-
+  condition_holds(Effect), condition_holds(Cause),
+  attr(Name, A1, A2, A3),
+  condition_requirement(Effect, Name, A1, A2, A3),
+  imposed_constraint(Cause, Name, A1, A2, A3).
+condition_cause(Effect, Cause) :-
+  condition_holds(Effect), condition_holds(Cause),
+  attr(Name, A1, A2, A3, A4),
+  condition_requirement(Effect, Name, A1, A2, A3, A4),
+  imposed_constraint(Cause, Name, A1, A2, A3, A4).
```

Comment:
Is this mapping complete? As far as I understand it maps imposed constraints (the cause) to condition requirements (the effect) which are exactly the same. So my understanding is that it would catch:
```
% cause
imposed_constraint(Cause,"node_version_satisfies","amdblis","3.0.1:").
% effect
condition_requirement(Effect,"node_version_satisfies","amdblis","3.0.1:").
```
but miss:
```
% cause
imposed_constraint(Cause,"node_version_satisfies","amdblis","3.0.1:").
% effect
condition_requirement(Effect,"node_version_satisfies","amdblis","2.0.0:").
```
? Is that correct, or am I missing some steps?